### PR TITLE
Extended Korean Filename Parsing Support

### PR DESCRIPTION
**Pull request by DeltaLaboratory**
# Added
- Added: Added -화, -회, -장 for volume parsing for Korean filenames
- Added: Added -회 for chapter parsing for Korean filenames

### CODE REVIEW

File: API/Services/Tasks/Scanner/Parser/Parser.cs [hunk 1]
```diff
@@ -557,7 +561,7 @@ public static class Parser
             MatchOptions, RegexTimeout),
         // Korean Chapter: 제n화 -> Chapter n, 가디언즈 오브 갤럭시 죽음의 보석.E0008.7화#44
         new Regex(
-            @"제?(?<Chapter>\d+\.?\d+)(화|장)",
+            @"제?(?<Chapter>\d+)(회|화|장)",
```
Comment (majora2007) [inline API/Services/Tasks/Scanner/Parser/Parser.cs:564]: Don't remove the support for decimal chapters

File: API/Services/Tasks/Scanner/Parser/Parser.cs [hunk 2]
```diff
@@ -119,6 +119,10 @@ public static class Parser
         new Regex(
             @"(卷|册)(?<Volume>\d+)",
             MatchOptions, RegexTimeout),
+        // Korean Volume: 제n화 -> Volume n, n화 -> Volume n
+        new Regex(
+            @"제?(?<Volume>\d+)(회|화|장)",
```
Comment (majora2007) [inline API/Services/Tasks/Scanner/Parser/Parser.cs:124]: Make this support decimal volumes whenever possible. aka 3.5 회

File: API.Tests/Parser/MangaParserTests.cs [hunk 3]
```diff
@@ -77,6 +78,9 @@ public MangaParserTests(ITestOutputHelper testOutputHelper)
     [InlineData("Манга Глава 2", "0")]
     [InlineData("Манга Тома 1-4", "1-4")]
     [InlineData("Манга Том 1-4", "1-4")]
+    [InlineData("조선왕조실톡 106화", "106")]
```
Comment (majora2007) [inline API.Tests/Parser/MangaParserTests.cs:81]: Try to keep all the Korean test cases together

**Review by majora2007 — CHANGES_REQUESTED** [review]
Please address my comments and add any new unit tests that may be required.

File: API/Services/Tasks/Scanner/Parser/Parser.cs (hunk 1 above)
Comment (DeltaLaboratory) [reply API/Services/Tasks/Scanner/Parser/Parser.cs]: Sorry. I was confused.
Fixed.

File: API/Services/Tasks/Scanner/Parser/Parser.cs (hunk 2 above)
Comment (DeltaLaboratory) [reply API/Services/Tasks/Scanner/Parser/Parser.cs]: Added Support.

Comment (DeltaLaboratory) [reply API/Services/Tasks/Scanner/Parser/Parser.cs]: Also I added unit tests for them

File: API.Tests/Parser/MangaParserTests.cs (hunk 3 above)
Comment (DeltaLaboratory) [reply API.Tests/Parser/MangaParserTests.cs:79]: Resolved.

File: API/Services/Tasks/Scanner/Parser/Parser.cs [hunk 4]
```diff
@@ -119,9 +119,9 @@ public static class Parser
         new Regex(
             @"(卷|册)(?<Volume>\d+)",
             MatchOptions, RegexTimeout),
-        // Korean Volume: 제n권 -> Volume n, n권  -> Volume n, 63권#200.zip -> Volume 63 (no chapter, #200 is just files inside)
+        // Korean Volume: 제n화 -> Volume n, n화 -> Volume n
```
Comment (majora2007) [inline API/Services/Tasks/Scanner/Parser/Parser.cs:122]: , 63권#200.zip -> Volume 63 (no chapter, #200 is just files inside)

I need this comment here, unless this isn't a valid case (but from last Korean requester, they mentioned this naming case is valid and 200 is not a chapter identifier)

**Review by majora2007 — COMMENTED** [review]
I just had one question for you, otherwise it looks good. Once we get that resolved, I can merge it.

File: API/Services/Tasks/Scanner/Parser/Parser.cs (hunk 4 above)
Comment (DeltaLaboratory) [reply API/Services/Tasks/Scanner/Parser/Parser.cs:122]: Ah Okay
It seems deleted when I merged -권 Pattern

Comment (DeltaLaboratory) [reply API/Services/Tasks/Scanner/Parser/Parser.cs]: Fixed

**Comment by DeltaLaboratory** [comment]
🮘🮙🮚🮛🮜🮝

**Review by majora2007 — APPROVED** [review]

**Pull request merged**